install-wings.sh

```diff
@@ -447,70 +390,85 @@ function firewall_ufw {
+  [ "$CONFIGURE_LETSENCRYPT" == true ] && firewall-cmd --add-service=http --permanent -q # Port 80
+  [ "$CONFIGURE_LETSENCRYPT" == true ] && firewall-cmd --add-service=https --permanent -q # Port 443
 
   firewall-cmd --permanent --zone=trusted --change-interface=pterodactyl0 -q
   firewall-cmd --zone=trusted --add-masquerade --permanent
-  firewall-cmd --ad-service=ssh --permanent -q # Port 22
   firewall-cmd --reload -q # Enable firewall
 
   echo "* Firewall-cmd installed"
   print_brake 70
 }
 
+letsencrypt() {
+  FAILED=false
+
+  # Install certbot
+  if [ "$OS" == "debian" ] || [ "$OS" == "ubuntu" ]; then
+    apt-get install certbot -y
+  elif [ "$OS" == "centos" ]; then
+    [ "$OS_VER_MAJOR" == "7" ] && yum install certbot
+    [ "$OS_VER_MAJOR" == "8" ] && dnf install certbot
+  else
+    # exit
+    print_error "OS not supported."
+    exit 1
+  fi
+
+  # If user has nginx
+  systemctl stop nginx || true
+
+  # Obtain certificate
+  certbot certonly --no-eff-email --email "$EMAIL" --standalone -d "$FQDN" || FAILED=true
+
+  systemctl start nginx || true
+
+  # Check if it succeded
+  if [ ! -d "/etc/letsencrypt/live/$FQDN/" ] || [ "$FAILED" == true ]; then
+    print_warning "The process of obtaining a Let's Encrypt certificate failed!"
+  fi
+}
+
 ####################
 ## MAIN FUNCTIONS ##
 ####################
-function perform_install {
+
+perform_install() {
   echo "* Installing pterodactyl wings.."
-  [ "$CONFIGURE_UFW" == true ] && firewall_ufw
-  [ "$CONFIGURE_FIREWALL_CMD" == true ] && firewall_firewalld
-  install_dep
+  [ "$OS" == "ubuntu" ] || [ "$OS" == "debian" ] && apt_update
+  [ "$OS" == "centos" ] || [ "$OS_VER_MAJOR" == "7" ] && yum_update
+  [ "$OS" == "centos" ] || [ "$OS_VER_MAJOR" == "8" ] && dnf_update
+  "$CONFIGURE_UFW" && firewall_ufw
+  "$CONFIGURE_FIREWALL_CMD" && firewall_firewalld
   install_docker
   ptdl_dl
   systemd_file
-  [ "$INSTALL_MARIADB" == true ] && install_mariadb
-  [ "$CONFIGURE_LETSENCRYPT" == true ] && letsencrypt
+  "$INSTALL_MARIADB" && install_mariadb
+  "$CONFIGURE_LETSENCRYPT" && letsencrypt
 
   # return true if script has made it this far
   return 0
 }
 
-ask_letsencrypt() {
-  if [ "$CONFIGURE_UFW" == false ] && [ "$CONFIGURE_FIREWALL_CMD" == false ]; then
-    print_warning "Let's Encrypt requires port 80/443 to be opened! You have opted out of the automatic firewall configuration; use this at your own risk (if port 80/443 is closed, the script will fail)!"
-  fi
-
-  print_warning "You cannot use Let's Encrypt with your hostname as an IP address! It must be a FQDN (e.g. node.example.org)."
-
-  echo -e -n "* Do you want to automatically configure HTTPS using Let's Encrypt? (y/N): "
-  read -r CONFIRM_SSL
-
-  if [[ "$CONFIRM_SSL" =~ [Yy] ]]; then
-    CONFIGURE_LETSENCRYPT=true
-  fi
-}
-
-function main {
+main() {
   # check if we can detect an already existing installation
-  if [ -d "/etc/pterodactyl" ]; then
+  if [ -d "/etc/pterodactyl/" ]; then
```

Comment:
Not too sure the reasoning behind this change?

Reply:
hmmmn copying mistake?

Reply:
Sorry, what do you mean?

Reply:
So I should remove it? It  doesn't make any difference after all.

Reply:
Well yes, it isn't more correct to do this way what?

Reply:
In my Linux experience I felt that having a `/` at the end made it look more as directory and not a file. It's just personal experience so I will change it. :+1: